### DISCUSSION TITLE
Cinnamon Workspaces

### DISCUSSION THREAD

**Discussion by rcalixte** [linuxmint/discussions#37] [Pull requests]
**Preface**
The following pull requests were opened in an effort to separate out key bindings and keyboard menu options for workspace up/down and workspace/window selection:
* linuxmint/cinnamon#10989
* linuxmint/cinnamon-desktop#214
* linuxmint/muffin#628

**Challenges**
The core challenges as correctly indicated by @mtwebster are two-fold:
* the changes proposed as-is would be disruptive to users (without being able to retain the current state, key bindings would effectively be reset so that they can be independent)
* the separation of workspace up/down key bindings begs the question of whether or not the core of Cinnamon should offer more native workspace management solutions (as opposed to Applets and/or Extensions)

**Ideas**
[This applet](https://github.com/linuxmint/cinnamon-spices-applets/tree/master/workspace-grid%40hernejj) contains what would effectively be combined with the current state to derive additional stock functionality. The release would be done in a way that user impact would be minimal, if at all. Ideally targeted for a major release of Linux Mint (e.g. 22.0) but communicated extremely well regardless so that users of other distros (Arch, Fedora, etc.) would not face disruption.

**Discussion**
While trying to focus on minimizing user impact and extending the current workspace functionality, the discussion is regarding the workspace functionality roadmap for Cinnamon and any possible improvements that can be done in that space. The goal would be a clear decision with a path forward for execution.

**Scope**
The intended scope of discussion here is not just configuration but workspace management in general. While these pull requests are focused on a limited scope, there are other aspects for inclusion, such as the multi-monitor behavior options observed here (borrowed from the current pre-alpha COSMIC):
 ![multi-monitor behavior](https://user-images.githubusercontent.com/106764128/200681731-7f5ecd59-222e-4935-aaba-ab84464c4d91.gif)

**Thanks**
Thanks to @mtwebster and @clefebvre for everything up to this point.

**Comment by corbin-auriti** [comment]
Sounds good! Separating key bindings for workspace management could be a bit disruptive, but it's a step in the right direction. Excited to see these improvements!

**Reply by adelpourmand** [reply]
At least I prefer i could assign different background for each workspace.

**Reply by rcalixte** [reply]
A different background for each workspace or for each display/monitor? If it's the latter, you can use [HydraPaper](https://gitlab.gnome.org/GabMus/HydraPaper) for that.

**Reply by adelpourmand** [reply]
As I tested Hydrapaper, it is suitable for different display/monitor and is not suitable for different workspace on one display/monitor.

**Reply by igorsantos07** [reply]
@adelpourmand, please open a new feature request! I would happily vote for it.

> PS: I used to run Nitrogen to use different wallpapers in each monitor, but nowadays I'm mostly happy with the refreshing take of a ~Bing~ daily wallpaper extension :eyes:

**Reply by ElectroArtifex** [reply]
> At least I prefer i could assign different background for each workspace.

[This script setup](https://forums.linuxmint.com/viewtopic.php?p=2453348#p2453348) still works for having different wallpapers and themes on each workspace. The only downside is that you must navigate the workspaces with the script hotkeys, so it doesn't integrate well with workspace switching applets and desklets. 
I'd love to see better visual separation and customization of individual workspaces become a native feature of Cinnamon.
